components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -951,6 +974,11 @@ void ESP8266::_oob_busy()
         MBED_ERROR(MBED_MAKE_ERROR(MBED_MODULE_DRIVER, MBED_ERROR_CODE_ENOMSG), \
                    "ESP8266::_oob_busy() AT timeout\n");
     }
+    if (!_busy)
```

Comment:
We may as well just abort the current command here, right? If it says busy we know it's not going to say "OK". It the outer loop wants to retry immediately it can check elapsed time and reconsider. Else fall back to EWOULDBLOCK and retry after timed sigio.

There's probably never any need to be waiting inside the driver itself, unless we really are expecting another command to work if we just wait a few milliseconds.

Reply:
If we abort here then do we need to set the 1/10s timeout?

Reply:
Timeouts would ideally be a "something's gone wrong" situation, and should never be hit. Timeout on recv("SEND OK") mattered before when we weren't detecting the BUSY, but if we now positively reckon we'll see BUSY or SEND OK or ERROR(?), timeout shouldn't really matter. Can be high if we never hit it in practice.